README.md

```diff
@@ -57,24 +73,29 @@ worker.get('/myResource.html', function(request, response) {
 });
 ```
 
-## Why I would like to support more than just handling requests?
+TODO: a working code example that can be executed to see how 'Your code' would actually look like
+
+### onInstall(?) TODO
+### onActivate(?) TODO
+
+### `onFetch(request, response)` returns Promise
 
-Because we could be interested on doing things during ServiceWorker installation (like precaching resources), clear caches (during activation, if the ServiceWorker changed), or we just want to communicate sending messages, remember at the end of the day we have code to handle request, and could be interesting to vary the content of caches or even ServiceWorker behaviour at will with a message.
+Will be called whenever the browser requests a resource when the worker has been installed.
 
-## Can I use more than one middleware combined?
+* `request`: it's a standard [Request Object](https://fetch.spec.whatwg.org/#concept-request)
+* `response`: it's a standard [Response Object](https://fetch.spec.whatwg.org/#concept-response)
 
-Right, you can do it, take into account that the request/response objects will be passing through them sequentially in the order that we register our middleware objetcs.
+You need to `.clone()` the `request` before using it as some fields are one use only.
 
-## Handling requests
+TODO: What is the use of the `response` parameter?
```

Comment:
After the whole process of iterating over the different middleware you need to return a Response object, different middleware could use the previous response object to perform any operations over content, headers or anything.